### ISSUE TITLE
Improve docs on Discriminated Unions

### ISSUE THREAD

**Issue by thany** [microsoft/TypeScript-Website#184]
Documentation link: https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions

The documentation explained three of the four important parts of this concept:
1. How to declare discrimination
2. How to make a discriminatable union
3. How to check the discrimination tag at runtime

But there's one thing missing: how to _instantiate_ a discriminatable union value.

It turns out, you'd have to add those discrimination tags at runtime. Personally I would expect that, following the docs, this would be correct:
```typescript
const shape: Shape = { size: 4 } as Square;
```
And that would be it. But it'll spew an error telling me I forgot to define the member `kind`.

I feel it's important that the docs explain why this is, and that it's required to write it like so:
```typescript
const shape: Shape = { kind: 'square', size: 4 };
```
And also that discrimination tags are indeed acessible at runtime, by javascript. In other words, it's not any kind of compiler magic. It's basically an interface member that can only ever have one value.

In short, the docs don't say anything about how to instantiate a value of a discriminated union type.

**Comment by orta** [comment]
I think we do a better job in the v2 handbook on discriminating unions: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions
